### DISCUSSION TITLE
Create a tool that will run black box tests on a cost-utility model

### DISCUSSION THREAD

**Discussion by CKENNEDY-HE** [HTAhackathon2024/.github#16] [Ideas]
### What is the problem we're trying to solve

Within a NICE STA submission, there is a "Model Validation" section near the end. As a member of the EAG we run our own model validation checks on the model. This may involve checking the survival estimates against registry data but it also includes general bug checking of the model. One of the generic criticisms of R models versus those written in Excel is the lack of transparency for predominantly excel modellers when moving to R.

It would be useful for modellers to be able to run these black box tests using an R package to not only save time but to maintain uniformity on the validation checks done on models submitted for NICE approval. It would also aid those who are most familiar with Excel models to understand that the model is firstly, working as it should and secondly, be able to find the variables that are informing the model outputs. 

Below is a link to the Tappenden and Chilcott 2014 paper which has the Black Box verification checks.
https://link.springer.com/article/10.1007/s40273-014-0186-2

### How will we tackle the problem?

I am hoping others could input on the roadmap here as I am not overly familiar with creating R packages, but I am learning! Just adding an idea to the discussion.

**Comment by n8thangreen** [comment]
How is this different form the unit tests specific to HTA Discussion idea?
